// Copyright 2015 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.
//
// Offline page item protocol for storage and exchanging of offline page
// metadata.

syntax = "proto2";

option optimize_for = LITE_RUNTIME;
option retain_unknown_fields = true;

package offline_pages;

message OfflinePageEntry {
  // URL of the offline page.
  required string url = 1;

  // Bookmark ID of the offline page.
  required int64 bookmark_id = 2;

  // Version of the offline page metadata.
  required int32 version = 3;

  // Path to the offline archive.
  required string file_path = 4;

  // Size of the offline archive.
  optional int64 file_size = 5;

  // Creation time of the offline archive.
  optional int64 creation_time = 6;

  // Last access time of the offline archive.
  optional int64 last_access_time = 7;

  // Number of times that the offline archive has been accessed.
  optional int32 access_count = 8;

  // Flags about the state and behavior of the offline page.
  enum Flags {
    // No flag is set.
    NONE = 0;
    // Indicates that the page is marked for deletion. The real deletion will
    // occur soon, after which the undo will not be possible.
    MARKED_FOR_DELETION = 1;
  };

  // Flags for the offline page.
  optional Flags flags = 9;
}
